lib/matplotlib/image.py

```diff
@@ -733,6 +735,31 @@ def set_data(self, A):
         self._rgbacache = None
         self.stale = True
 
+    def _convert_units(self, A):
+        # Take the first element since units expects a 1D sequence, not 2D
+        converter = munits.registry.get_converter(A[0])
```

Comment:
We probably want to stash the converter the first time through?

Reply:
Makes sense. It does raise the question of making converter immutable once it has been set once (ie. the first time `set_data()` is called). Should this be the case, or should we provide a method to manually change the converter after a `set_data()` call?